src/libostree/ostree-repo.c

```diff
@@ -4830,10 +4829,10 @@ ostree_repo_add_gpg_signature_summary (OstreeRepo     *self,
                       cancellable, error))
         return FALSE;
 
-      new_metadata = _ostree_detached_metadata_append_gpg_sig (existing_signatures, signature_data);
+      metadata = _ostree_detached_metadata_append_gpg_sig (metadata, signature_data);
```

Comment:
One subtle issue here is that we're leaking the previous value of `metadata`. Let's tweak it to something like this instead:

```c
g_autoptr(GVariant) old_metadata = g_steal_pointer (&metadata);
metadata = _ostree_detached_metadata_append_gpg_sig (old_metadata, signature_data);
```

?